packages/headless/src/features/search/search-actions.ts

```diff
@@ -94,18 +95,16 @@ const fetchFromAPI = async (
 
 /**
  * Executes a search query.
- * @param analyticsAction (SearchAction) The analytics action to log after a successful query.
+ * @param analyticsAction - The analytics action to log after a successful query.
  */
-export const executeSearch = createAsyncThunk<
+export const executeSearch: AsyncThunkDefinition<
+  'search/executeSearch',
```

Comment:
For the action name to be read, it needs to be part of the type. The value of `executeSearch` const is lost in the `.d.ts` file needed by the API extractor.

Reply:
Personally, I feel we should not document the action `type` string, since intellisense shows the variable/function name, not the string it is holding. I know you wanted to use this to organize the docs, but maybe that could be done with the configuration file instead?

Reply:
Since the talk about refactoring action namespaces, I don't think that's necessary anymore.

However, I believe the type string would be useful for people using the redux dev tools.

Reply:
Update: for this PR, I'll remove the action type string parsing.